test-lit/Fir/fir-ops.fir

```diff
@@ -2,69 +2,14 @@
 
 // RUN: tco -emit-fir %s | tco -emit-fir | FileCheck %s
 
-// Fortran Intrinsic types
 // CHECK-LABEL: func @it1() -> !fir.int<4>
-// CHECK-LABEL: func @it2() -> !fir.real<8>
-// CHECK-LABEL: func @it3() -> !fir.complex<8>
-// CHECK-LABEL: func @it4() -> !fir.logical<1>
-// CHECK-LABEL: func @it5() -> !fir.char<1>
 func @it1() -> !fir.int<4>
-func @it2() -> !fir.real<8>
-func @it3() -> !fir.complex<8>
-func @it4() -> !fir.logical<1>
-func @it5() -> !fir.char<1>
-
-// Fortran Derived types (records)
-// CHECK-LABEL: func @dvd1() -> !fir.type<derived1>
-// CHECK-LABEL: func @dvd2() -> !fir.type<derived2(p:i32)>
-// CHECK-LABEL: func @dvd3() -> !fir.type<derived3{f:f32}>
-// CHECK-LABEL: func @dvd4() -> !fir.type<derived4(p:i8){f:f32}>
-func @dvd1() -> !fir.type<derived1>
-func @dvd2() -> !fir.type<derived2(p:i32)>
-func @dvd3() -> !fir.type<derived3{f:f32}>
-func @dvd4() -> !fir.type<derived4(p:i8){f:f32}>
-
-// FIR array types
-// CHECK-LABEL: func @arr1() -> !fir.array<10xf32>
-// CHECK-LABEL: func @arr2() -> !fir.array<10x10xf32>
-// CHECK-LABEL: func @arr3() -> !fir.array<?xf32>
-// CHECK-LABEL: func @arr4() -> !fir.array<10x?xf32>
-// CHECK-LABEL: func @arr5() -> !fir.array<?x?xf32>
-// CHECK-LABEL: func @arr6() -> !fir.array<*:f32>
-func @arr1() -> !fir.array<10xf32>
-func @arr2() -> !fir.array<10x10xf32>
-func @arr3() -> !fir.array<?xf32>
-func @arr4() -> !fir.array<10x?xf32>
-func @arr5() -> !fir.array<?x?xf32>
-func @arr6() -> !fir.array<*:f32>
-
-// FIR pointer-like types
-// CHECK-LABEL: func @mem1() -> !fir.ref<i32>
-// CHECK-LABEL: func @mem2() -> !fir.ptr<i32>
-// CHECK-LABEL: func @mem3() -> !fir.heap<i32>
-// CHECK-LABEL: func @mem4() -> !fir.ref<() -> ()>
-func @mem1() -> !fir.ref<i32>
-func @mem2() -> !fir.ptr<i32>
-func @mem3() -> !fir.heap<i32>
-func @mem4() -> !fir.ref<() -> ()>
-
-// FIR box types (descriptors)
-// CHECK-LABEL: func @box1() -> !fir.box<!fir.array<?xf32>>
-// CHECK-LABEL: func @box2() -> !fir.boxchar<2>
-// CHECK-LABEL: func @box3() -> !fir.boxproc<(i32, i32) -> i64>
-// CHECK-LABEL: func @box4() -> !fir.box<!fir.type<derived3{f:f32}>>
-func @box1() -> !fir.box<!fir.array<?xf32>>
-func @box2() -> !fir.boxchar<2>
-func @box3() -> !fir.boxproc<(i32, i32) -> i64>
-func @box4() -> !fir.box<!fir.type<derived3{f:f32}>>
-
-// FIR misc. types
-// CHECK-LABEL: func @oth1() -> !fir.dims<1>
-// CHECK-LABEL: func @oth2() -> !fir.field
-// CHECK-LABEL: func @oth3() -> !fir.tdesc<!fir.type<derived5{f1:f32,f2:f32}>>
-func @oth1() -> !fir.dims<1>
-func @oth2() -> !fir.field
-func @oth3() -> !fir.tdesc<!fir.type<derived5{f1:f32,f2:f32}>>
+// CHECK-LABEL: func @box1() -> !fir.boxchar<2>
+func @box1() -> !fir.boxchar<2>
+// CHECK-LABEL: func @box2() -> !fir.boxproc<(i32, i32) -> i64>
+func @box2() -> !fir.boxproc<(i32, i32) -> i64>
+// CHECK-LABEL: func @box3() -> !fir.box<!fir.type<derived3{f:f32}>>
+func @box3() -> !fir.box<!fir.type<derived3{f:f32}>>
```

Comment:
Are you just removing the unreferenced func ops?

Reply:
yes, I removed all the redundant test but these four because the related functions appear in fir.call.